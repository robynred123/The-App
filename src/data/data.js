 export const data = [
        {
            "key": 1,
            "point": "Existing Knowledge of Products in Development"
        },
        {
            "key": 2,
            "point": "Already proven a good fit with the team"
        },
        {
            "key": 3,
            "point": "Good knowledge of languages and frameworks used by the team, e.g. React, Node.js"
        },
        {
            "key": 4,
            "point": "Eager to get involved, and to learn"
        },
        {
            "key": 5,
            "point": "Good problem solving skills"
        },
        {
            "key": 6,
            "point": "2 years development experience in Asda, having worked on a large scale application deployed to a number of stores across the country"
        }
    ]