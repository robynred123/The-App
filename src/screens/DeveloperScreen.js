import React from 'react';
import { StyleSheet, Text, View, ScrollView } from 'react-native';
import { TextCarousel } from '../components/TextCarousel';
import { data } from '../data/data';

const DeveloperScreen = () => {
  return (
    <ScrollView>
    <View>

      <Text style={styles.title}>What Can I Bring To Asda?</Text>
    
      <TextCarousel 
        data={data} 
      />

    </View>
    <View>
      <Text style={styles.title}>Why Do I Want to Be A Developer?</Text>
      <Text style={styles.text}>
          Before beginning my apprenticeship I felt that I had reached a wall in my ability to improve in my previous job, and I was looking for a
          position where I could continuously learn. At Asda I have been introduced to a lot of new experiences, 
          and have been given the opportunity to continue to improve my skills,
          and learn about new technologies. 
      </Text>
      <Text style={styles.text}>
          I want to continue to work as a developer to continue this learning and to be able to make applications that people will enjoy using. 
          In addition I want to work towards encouraging other females to get into technology and specifically development, 
          as I feel that when I was considering career options, it wasn't 
          something that I was made aware of, despite my keen interest in technology.    
      </Text>
    </View>
    </ScrollView>
  );
}
    
const styles = StyleSheet.create({
  title: {
    fontSize: 20,
    fontWeight: '700',
    justifyContent: 'center',
    textAlign: 'center',
    padding: 5
  },
  text: {
    padding: 10,
    fontSize: 15,
    flexWrap: 'wrap',
    textAlign: 'center'
  }
});

export default DeveloperScreen