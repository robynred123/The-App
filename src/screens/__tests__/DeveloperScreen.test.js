import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import DeveloperScreen from '../DeveloperScreen';
import { data } from '../../data/data';

Enzyme.configure({ adapter: new Adapter() })

describe('<DeveloperScreen />', () => {
    it('should render the DeveloperScreen without crashing', () => {
        const component = shallow(<DeveloperScreen />)
        expect(component).toMatchSnapshot()
    });

    it('should render the DeveloperScreen with text', () => {
        const component = shallow(<DeveloperScreen />)
        expect(component.find('Text').first().children().text()).toEqual('What Can I Bring To Asda?')
        expect(component).toMatchSnapshot()
    });

    it('should render the DeveloperScreen with text', () => {
        const component = shallow(<DeveloperScreen />)
        expect(component.find('TextCarousel').props().data).toEqual(data)
        expect(component).toMatchSnapshot()
    });
})