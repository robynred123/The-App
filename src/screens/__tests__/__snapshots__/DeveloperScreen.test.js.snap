// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DeveloperScreen /> should render the DeveloperScreen with text 1`] = `
<ScrollViewMock>
  <View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "700",
          "justifyContent": "center",
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      What Can I Bring To Asda?
    </Text>
    <TextCarousel
      data={
        Array [
          Object {
            "key": 1,
            "point": "Existing Knowledge of Products in Development",
          },
          Object {
            "key": 2,
            "point": "Already proven a good fit with the team",
          },
          Object {
            "key": 3,
            "point": "Good knowledge of languages and frameworks used by the team, e.g. React, Node.js",
          },
          Object {
            "key": 4,
            "point": "Eager to get involved, and to learn",
          },
          Object {
            "key": 5,
            "point": "Good problem solving skills",
          },
          Object {
            "key": 6,
            "point": "2 years development experience in Asda, having worked on a large scale application deployed to a number of stores across the country",
          },
        ]
      }
    />
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "700",
          "justifyContent": "center",
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
       Why Do I Want to Be A Developer?
    </Text>
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 15,
          "padding": 10,
          "textAlign": "center",
        }
      }
    >
      Before beginning my apprenticeship I felt that I had reached a wall in my ability to improve in my previous job, and I was looking for a position where I could continuously learn. At Asda I have been introduced to a lot of new experiences, and have been given the opportunity to continue to improve my skills, and learn about new technologies.
    </Text>
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 15,
          "padding": 10,
          "textAlign": "center",
        }
      }
    >
      I want to continue to work as a developer to continue this learning and to be able to make applications that people will enjoy using. In addition I want to work towards encouraging other females to get into technology and specifically development, as I feel that when I was considering career options, it wasn't something that I was made aware of, despite my keen interest in technology.
    </Text>
  </View>
</ScrollViewMock>
`;

exports[`<DeveloperScreen /> should render the DeveloperScreen with text 2`] = `
<ScrollViewMock>
  <View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "700",
          "justifyContent": "center",
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      What Can I Bring To Asda?
    </Text>
    <TextCarousel
      data={
        Array [
          Object {
            "key": 1,
            "point": "Existing Knowledge of Products in Development",
          },
          Object {
            "key": 2,
            "point": "Already proven a good fit with the team",
          },
          Object {
            "key": 3,
            "point": "Good knowledge of languages and frameworks used by the team, e.g. React, Node.js",
          },
          Object {
            "key": 4,
            "point": "Eager to get involved, and to learn",
          },
          Object {
            "key": 5,
            "point": "Good problem solving skills",
          },
          Object {
            "key": 6,
            "point": "2 years development experience in Asda, having worked on a large scale application deployed to a number of stores across the country",
          },
        ]
      }
    />
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "700",
          "justifyContent": "center",
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
       Why Do I Want to Be A Developer?
    </Text>
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 15,
          "padding": 10,
          "textAlign": "center",
        }
      }
    >
      Before beginning my apprenticeship I felt that I had reached a wall in my ability to improve in my previous job, and I was looking for a position where I could continuously learn. At Asda I have been introduced to a lot of new experiences, and have been given the opportunity to continue to improve my skills, and learn about new technologies.
    </Text>
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 15,
          "padding": 10,
          "textAlign": "center",
        }
      }
    >
      I want to continue to work as a developer to continue this learning and to be able to make applications that people will enjoy using. In addition I want to work towards encouraging other females to get into technology and specifically development, as I feel that when I was considering career options, it wasn't something that I was made aware of, despite my keen interest in technology.
    </Text>
  </View>
</ScrollViewMock>
`;

exports[`<DeveloperScreen /> should render the DeveloperScreen without crashing 1`] = `
<ScrollViewMock>
  <View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "700",
          "justifyContent": "center",
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
      What Can I Bring To Asda?
    </Text>
    <TextCarousel
      data={
        Array [
          Object {
            "key": 1,
            "point": "Existing Knowledge of Products in Development",
          },
          Object {
            "key": 2,
            "point": "Already proven a good fit with the team",
          },
          Object {
            "key": 3,
            "point": "Good knowledge of languages and frameworks used by the team, e.g. React, Node.js",
          },
          Object {
            "key": 4,
            "point": "Eager to get involved, and to learn",
          },
          Object {
            "key": 5,
            "point": "Good problem solving skills",
          },
          Object {
            "key": 6,
            "point": "2 years development experience in Asda, having worked on a large scale application deployed to a number of stores across the country",
          },
        ]
      }
    />
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 20,
          "fontWeight": "700",
          "justifyContent": "center",
          "padding": 5,
          "textAlign": "center",
        }
      }
    >
       Why Do I Want to Be A Developer?
    </Text>
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 15,
          "padding": 10,
          "textAlign": "center",
        }
      }
    >
      Before beginning my apprenticeship I felt that I had reached a wall in my ability to improve in my previous job, and I was looking for a position where I could continuously learn. At Asda I have been introduced to a lot of new experiences, and have been given the opportunity to continue to improve my skills, and learn about new technologies.
    </Text>
    <Text
      style={
        Object {
          "flexWrap": "wrap",
          "fontSize": 15,
          "padding": 10,
          "textAlign": "center",
        }
      }
    >
      I want to continue to work as a developer to continue this learning and to be able to make applications that people will enjoy using. In addition I want to work towards encouraging other females to get into technology and specifically development, as I feel that when I was considering career options, it wasn't something that I was made aware of, despite my keen interest in technology.
    </Text>
  </View>
</ScrollViewMock>
`;
